gamemodes/terrortown/gamemode/shared/sh_main.lua

```diff
@@ -10,6 +10,137 @@ local MAX_DROWN_TIME = 8
 
 local sneakSpeedSquared = math.pow(150, 2)
 
+isShopFallbackInitialized = false
+
+---
+-- Initializes the equipment with necessary data for ttt2
+-- @internal
+-- @realm shared
+local function TTT2PreRegisterSWEP(equipment, name)
+	if isShopFallbackInitialized then
+		MsgN("[TTT2] Trying to hotreload ",  name, " .")
+	end
+
+	-- Initialize Equipment
+	AddStandardKeyValues(equipment, name)
+	ShopEditor.InitDefaultData(equipment)
+
+	if isShopFallbackInitialized then
+		local oldSWEP = weapons.GetStored(name)
+
+		-- Keep custom changed data from the old SWEP if hotReloadableKeys are given
+		if #equipment.HotReloadableKeys > 0 and oldSWEP then
+			MsgN("[TTT2] Hotreloading ",  #equipment.HotReloadableKeys, " given Keys from old SWEP-file.")
+
+			for _, keys in pairs(equipment.HotReloadableKeys) do
+				local eqKeyField = equipment
+				local oldKeyField = oldSWEP
+				local keyString = ""
+
+				-- If only a single key is given, not a table, convert it
+				if isstring(keys) then
+					keys = {keys}
+				end
+
+				if not istable(keys) then continue end
+
+				local continueOuterLoop = false
+				local counter = 0
+				local saveKey = keys[1]
+
+				for _, key in pairs(keys) do
+					if not isstring(key) or not oldKeyField then
+						continueOuterLoop = true
+						break
+					end
+
+
+					keyString = keyString .. "." .. key
+					counter = counter + 1
+					eqKeyField[key] = eqKeyField[key] or {}
+					oldKeyField = oldKeyField[key]
+
+					-- To keep eqKeyField as reference check for tables or create one
+					if counter < #keys and not istable(eqKeyField[key]) then
+						eqKeyField[key] = {}
+					elseif counter == #keys then
+						saveKey = key
+						break
+					end
+
+					eqKeyField = eqKeyField[key]
+				end
+
+				if continueOuterLoop then continue end
+
+				MsgN("[TTT2] Overwriting SWEP",  keyString, " = ", tostring(eqKeyField[saveKey]), " with ", tostring(oldKeyField))
+
+				eqKeyField[saveKey] = oldKeyField
+			end
+		end
+
+		ResetDefaultEquipment(equipment)
+	end
+
+	if SERVER and sql.CreateSqlTable("ttt2_items", ShopEditor.savingKeys) then
+		if name == "weapon_ttt_demonicsheep" then print("3") end
```

Comment:
do not put into single line, also add an empty line after the end, the same holds true for the following lines

Reply:
oops, thats a leftover debug statement ^^

Reply:
now that I look at it again, it seems rather obvious that it is. :D